File: python/course-repo/exercise/01-skip-the-workflow/src/webhooks/activities.py
import httpx
from temporalio import activity

from .shared import WebhookDelivery


@activity.defn
def deliver_webhook(req: WebhookDelivery) -> int:
    activity.logger.info(
        "Delivering webhook for event %s to %s", req.event_id, req.url
    )
    # TODO: POST req.payload to req.url using httpx.post()
    # TODO: raise on non-2xx response (response.raise_for_status())
    # TODO: return the HTTP status code
    raise NotImplementedError("Fill in deliver_webhook")

File: typescript/course-repo/exercise/01-durable-job-queue/src/activities.ts
import { log } from '@temporalio/activity';
import type { WebhookDelivery } from './shared';

// Same activity whether this runs standalone or as a step inside a Workflow.
export async function deliverWebhook(req: WebhookDelivery): Promise<number> {
  log.info('Delivering webhook for event', { eventId: req.eventId, url: req.url });
  // TODO: POST req.payload to req.url using fetch()
  // TODO: throw if the response is not ok (response.status >= 400)
  // TODO: return the HTTP status code
  throw new Error('Fill in deliverWebhook');
}
